### PR TITLE
Add support for stripped go binaries

### LINKED ISSUE #428: Gotls Mode fails for stripped go binaries
**Describe the bug**

Currently `gotls mode` of ecapture doesn't supports stripped go binaries. The ecapture fails to intialise if stripped go binary is supplied in `--elfpath` flag.

**To Reproduce**

**pre-requisite**: any stripped go binary is required. I have used `containerd` for reproduction

1. Assuming containerd is present.  Check if it stripped
```sh
file /usr/bin/containerd
```

> /usr/bin/containerd: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=88f1113b377f304d1f31f0d979efe8ff4fcd85f6, for GNU/Linux 3.2.0, **stripped**

2. Perform the below command.
```sh
ecapture gotls --elfpath="/usr/bin/containerd"
```
3. You will error similar to below one
 > EBPFProbeGoTLS	module initialization failed. [skip it]. error:no symbol section

**Expected behavior**
gotls mode should be able to handle stripped go binary files.

**Additional context**

When ecapture with gotls command is executed, it perform various [checks](https://github.com/gojue/ecapture/blob/master/user/config/config_gotls.go#L54). 

In the last step it tries to find the retOffsets of `crypto/tls.(*Conn).Read` symbol at [here](https://github.com/gojue/ecapture/blob/master/user/config/config_gotls.go#L96)

In [`findRetOffsets() method`](https://github.com/gojue/ecapture/blob/master/user/config/config_gotls.go#L107-L120),  only `static symbols` are loaded and search is performed for `crypto/tls.(*Conn).Read`. But `static symbols` are not present in  stripped binaries. That is the reason it fails for stripped binaries

The PR #426, adds `dynamic symbols` in search space to fix the bug.

/cc @cfc4n

### PR DESCRIPTION
**Pull request by h0x0er**
Currently ecapture fails for stripped go binaries.

Because in `findRetOffsets()` method, only symbols from `sym table section` are consider for searching of `GoTlsReadFunc symbol`. Since in case of stripped go binaries no `sym table section` is present, ecapture fails to initialise.

This PR adds dynamic symbols in search space to fix the issue.

fixes #428

### CODE REVIEW

**Comment by h0x0er** [comment]
@cfc4n , I had created an issue with extra info. Thanks

File: user/config/config_gotls.go [hunk 1]
```diff
@@ -103,15 +104,25 @@ func (gc *GoTLSConfig) Check() error {
...
-	if err != nil {
-		return nil, err
+	var allSymbs []elf.Symbol
+
+	goSymbs, _ := gc.goElf.Symbols()
+	if len(goSymbs) > 0 {
+		allSymbs = append(allSymbs, goSymbs...)
+	}
+
+	goDynamicSymbs, _ := gc.goElf.DynamicSymbols()
```
Comment (cfc4n) [inline user/config/config_gotls.go:114]: in stripped compiled files ,  there are also no symbols in the `DynamicSymbols` section.

**Review by cfc4n — COMMENTED** [review]
I think this PR cannot support stripped Go binaries.

File: user/config/config_gotls.go (hunk 1 above)
Comment (h0x0er) [reply user/config/config_gotls.go:114]: @cfc4n , Thanks for commenting. I had shared more info in the #428 , for the reason behind including dynamic symbols

**Review by cfc4n — APPROVED** [review]
LGTM.

**Pull request merged**